web/_js/draw.js

```diff
@@ -384,6 +405,230 @@ function initDraw(){
+		})
+		periodDeleteEl.addEventListener('click', () => {
+			if (pathWithPeriods.length === 1) return
+			pathWithPeriods = pathWithPeriods.filter((_, i) => i !== index)
+			initPeriodGroups()
+		})
+		periodDuplicateEl.addEventListener('click', () => {
+			pathWithPeriods.push([pathWithPeriods[index][0], [...pathWithPeriods[index][1]]])
+			initPeriodGroups()
+		})
+
+		periodGroups.appendChild(periodGroupEl)
+		periodGroupElements.push({
+			periodGroupEl,
+			startPeriodEl,
+			endPeriodEl,
+			periodVisibilityEl
+		})
+	})
+	console.log(periodGroupTemplate)
+	updatePeriodGroups()
+
+}
+
+function updatePeriodGroups() {
+	var pathToActive = []
+	var lastActivePathIndex
+	var currentActivePathIndex 
+
+	periodGroupElements.forEach((elements, index) => {
+		let {
+			periodGroupEl,
+			startPeriodEl,
+			endPeriodEl,
+			periodVisibilityEl
+		} = elements
+
+		if (periodGroupEl.dataset.active === "true") lastActivePathIndex = index
+
+		if (isOnPeriod(
+			parseInt(startPeriodEl.value), 
+			parseInt(endPeriodEl.value),
+			period
+		)) {
+			pathToActive = pathWithPeriods[index][1]
+			currentActivePathIndex = index
+			periodGroupEl.dataset.active = true
+		} else {
+			periodGroupEl.dataset.active = false
+		}
+
+		pathWithPeriods[index][0] = formatPeriod(
+			parseInt(startPeriodEl.value), 
+			parseInt(endPeriodEl.value),
+			period
+		)
+	})
+
+	periodsStatus.textContent = ""
+	if (lastActivePathIndex !== undefined) {
+		if (lastActivePathIndex === currentActivePathIndex) {
+			// just update the path
+			pathWithPeriods[currentActivePathIndex] = [
+				formatPeriod(
+					parseInt(periodGroupElements[currentActivePathIndex].startPeriodEl.value),
+					parseInt(periodGroupElements[currentActivePathIndex].endPeriodEl.value)
+				),
+				path
+			]
+			console.log(pathWithPeriods[currentActivePathIndex])
+		} else if (currentActivePathIndex === undefined) {
+			pathWithPeriods[lastActivePathIndex][1] = path
+			updatePath([])
+			periodsStatus.textContent = "No paths available on this period!"
+		} else {
+			// switch the path
+			pathWithPeriods[lastActivePathIndex][1] = path
+			updatePath(pathToActive)
+
+		}
+	} else {
+		updatePath(pathToActive)
+	}
+
+	drawing = disableDrawingOverride ? false : currentActivePathIndex !== undefined
+	
+}
+
+function parsePeriod(periodString) {
+	periodString = periodString + ""
+	// TODO: Support for multiple/alternative types of canvas
+	if (periodString.search('-') + 1) {
+		var [start, end] = periodString.split('-').map(i => parseInt(i))
+		return [start, end]
+	} else {
+		let periodNew = parseInt(periodString)
+		return [periodNew, periodNew]
+	}
+}
+
+function formatPeriod(start, end) {
+	if (start === end) return start
+	else return start + "-" + end
+}
 
+function updatePath(newPath = path) {
+	path = newPath
+	render(path)
+	finishButton.disabled = path.length;
+	undoButton.disabled = path.length == 1; // Maybe make it undo the cancel action in the future
```

Comment:
This prevents undoing the first point, which only leaves `Reset` to clear the first point

Reply:
Whoops. Fixed on the next push!